pkg/sql/catalog/replication/replication_test: TestReaderCatalogTSAdvance failed

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18895472?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18895472?buildTab=artifacts#/) on master @ [ea364d6cdd98b4f8d07be5e7928743f2c20b3a7e](https://github.com/cockroachdb/cockroach/commits/ea364d6cdd98b4f8d07be5e7928743f2c20b3a7e):


```
I250221 18:34:21.216284 27180 kv/kvserver/replica_consistency.go:210  [T1,Vsystem,n1,consistencyChecker,s1,r4/1:/System{/tsd-tse}] 2817  triggering stats recomputation to resolve delta of {ContainsEstimates:16640 LastUpdateNanos:1740162854179448216 LockAge:0 GCBytesAge:0 LiveBytes:-341510 LiveCount:-5512 KeyBytes:-323518 KeyCount:-5512 ValBytes:-17992 ValCount:-5512 IntentBytes:0 IntentCount:0 LockBytes:0 LockCount:0 RangeKeyCount:0 RangeKeyBytes:0 RangeValCount:0 RangeValBytes:0 SysBytes:0 SysCount:0 AbortSpanBytes:0}
I250221 18:34:21.504908 27650 sql/catalog/lease/descriptor_state.go:135  [n1] 2818  new lease: tabledesc.immutable: {ID: 4, Version: 14, ModificationTime: "1740162861.445309874,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 18:34:21.506186 18733 sql/catalog/lease/lease.go:480  [-] 2819  waiting for 1 leases to expire: desc=[{users 4 13}]
I250221 18:34:21.506161 27642 sql/catalog/lease/descriptor_state.go:135  [n1] 2820  new lease: tabledesc.immutable: {ID: 20, Version: 14, ModificationTime: "1740162861.445309874,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 18:34:21.507487 27573 sql/catalog/lease/descriptor_state.go:135  [n1] 2821  new lease: tabledesc.immutable: {ID: 23, Version: 14, ModificationTime: "1740162861.445309874,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 18:34:21.508812 27387 sql/catalog/lease/descriptor_state.go:135  [n1] 2822  new lease: tabledesc.immutable: {ID: 33, Version: 14, ModificationTime: "1740162861.445309874,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 18:34:21.509989 27683 sql/catalog/lease/descriptor_state.go:135  [n1] 2823  new lease: tabledesc.immutable: {ID: 44, Version: 14, ModificationTime: "1740162861.445309874,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 18:34:21.511539 27688 sql/catalog/lease/descriptor_state.go:135  [n1] 2824  new lease: tabledesc.immutable: {ID: 104, Version: 13, ModificationTime: "1740162861.445309874,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 18:34:21.512789 27693 sql/catalog/lease/descriptor_state.go:135  [n1] 2825  new lease: tabledesc.immutable: {ID: 107, Version: 13, ModificationTime: "1740162861.445309874,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 18:34:21.514050 27697 sql/catalog/lease/descriptor_state.go:135  [n1] 2826  new lease: tabledesc.immutable: {ID: 109, Version: 13, ModificationTime: "1740162861.445309874,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 18:34:21.525011 27717 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38160,hostssl,user=reader] 2827 ={"Timestamp":1740162861525006933,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38160","SessionID":"18264ce71c067cb40000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 18:34:21.528731 27671 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38160,hostssl,user=reader] 2828 ={"Timestamp":1740162861528725928,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38160","SessionID":"18264ce71c067cb40000000000000001","Duration":12103832}
I250221 18:34:22.200105 28094 sql/catalog/lease/descriptor_state.go:135  [n1] 2829  new lease: tabledesc.immutable: {ID: 4, Version: 15, ModificationTime: "1740162862.156231282,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 18:34:22.201884 28378 sql/catalog/lease/descriptor_state.go:135  [n1] 2830  new lease: tabledesc.immutable: {ID: 20, Version: 15, ModificationTime: "1740162862.156231282,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 18:34:22.204296 28437 sql/catalog/lease/descriptor_state.go:135  [n1] 2831  new lease: tabledesc.immutable: {ID: 23, Version: 15, ModificationTime: "1740162862.156231282,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 18:34:22.206663 28407 sql/catalog/lease/descriptor_state.go:135  [n1] 2832  new lease: tabledesc.immutable: {ID: 33, Version: 15, ModificationTime: "1740162862.156231282,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 18:34:22.208314 28412 sql/catalog/lease/descriptor_state.go:135  [n1] 2833  new lease: tabledesc.immutable: {ID: 44, Version: 15, ModificationTime: "1740162862.156231282,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 18:34:22.209902 28417 sql/catalog/lease/descriptor_state.go:135  [n1] 2834  new lease: tabledesc.immutable: {ID: 104, Version: 14, ModificationTime: "1740162862.156231282,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250221 18:34:22.212152 28486 sql/catalog/lease/descriptor_state.go:135  [n1] 2835  new lease: tabledesc.immutable: {ID: 107, Version: 14, ModificationTime: "1740162862.156231282,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 18:34:22.213723 28491 sql/catalog/lease/descriptor_state.go:135  [n1] 2836  new lease: tabledesc.immutable: {ID: 109, Version: 14, ModificationTime: "1740162862.156231282,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250221 18:34:22.251359 28551 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38162,hostssl,user=reader] 2837 ={"Timestamp":1740162862251354798,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38162","SessionID":"18264ce74756c8070000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 18:34:22.253815 28549 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38162,hostssl,user=reader] 2838 ={"Timestamp":1740162862253809331,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38162","SessionID":"18264ce74756c8070000000000000001","Duration":10505200}
I250221 18:34:22.898931 18733 sql/catalog/lease/lease.go:480  [-] 2839  waiting for 1 leases to expire: desc=[{users 4 15}]
I250221 18:34:22.899912 28957 sql/catalog/lease/descriptor_state.go:135  [n1] 2840  new lease: tabledesc.immutable: {ID: 4, Version: 16, ModificationTime: "1740162862.863016796,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 18:34:22.903411 28920 sql/catalog/lease/descriptor_state.go:135  [n1] 2841  new lease: tabledesc.immutable: {ID: 20, Version: 16, ModificationTime: "1740162862.863016796,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 18:34:22.904972 28973 sql/catalog/lease/descriptor_state.go:135  [n1] 2842  new lease: tabledesc.immutable: {ID: 23, Version: 16, ModificationTime: "1740162862.863016796,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 18:34:22.907213 29168 sql/catalog/lease/descriptor_state.go:135  [n1] 2843  new lease: tabledesc.immutable: {ID: 33, Version: 16, ModificationTime: "1740162862.863016796,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 18:34:22.908879 29221 sql/catalog/lease/descriptor_state.go:135  [n1] 2844  new lease: tabledesc.immutable: {ID: 44, Version: 16, ModificationTime: "1740162862.863016796,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 18:34:22.910328 29226 sql/catalog/lease/descriptor_state.go:135  [n1] 2845  new lease: tabledesc.immutable: {ID: 104, Version: 15, ModificationTime: "1740162862.863016796,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f589f0]

goroutine 29212 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400f4e67a8?, {0x68907b8, 0x4008447d10})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a1ed20?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x529ec51?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x68907b8, 0x4008447d10})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x68907b8?, 0x4008447d10?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x401073f4d0, {0x68907b8, 0x4008447d10}, {0x40021a72f0, 0x11}, 0x400f4e6f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400d1f29f0, {0x68937f8?, 0x40042354a0?}, 0x4003dfc420, {0x400b3d0c98, 0x3}, {0x401073f4d0?, 0x54?}, 0x40057e1c20)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14857
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48034

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18896440?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18896440?buildTab=artifacts#/) on master @ [b9db2fde41bd0d0d03afe762bd2637d2542bf92b](https://github.com/cockroachdb/cockroach/commits/b9db2fde41bd0d0d03afe762bd2637d2542bf92b):


```
I250221 19:40:17.395469 36832 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52432,hostssl,user=reader] 3317 ={"Timestamp":1740166817395464887,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52432","SessionID":"18265080287cb2a50000000000000001","Duration":4890235}
I250221 19:40:17.408124 36845 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52434,hostssl,user=reader] 3318 ={"Timestamp":1740166817408120500,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52434","SessionID":"182650802959c3650000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.410298 36843 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52434,hostssl,user=reader] 3319 ={"Timestamp":1740166817410293026,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52434","SessionID":"182650802959c3650000000000000001","Duration":5230218}
I250221 19:40:17.424755 36899 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52436,hostssl,user=reader] 3320 ={"Timestamp":1740166817424750922,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52436","SessionID":"182650802a57143c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.435596 36817 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52436,hostssl,user=reader] 3321 ={"Timestamp":1740166817435590711,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52436","SessionID":"182650802a57143c0000000000000001","Duration":13925283}
I250221 19:40:17.451449 36930 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52438,hostssl,user=reader] 3322 ={"Timestamp":1740166817451444755,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52438","SessionID":"182650802bbaeb4a0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.453525 36709 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52438,hostssl,user=reader] 3323 ={"Timestamp":1740166817453520372,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52438","SessionID":"182650802bbaeb4a0000000000000001","Duration":8536236}
I250221 19:40:17.466114 36902 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52440,hostssl,user=reader] 3324 ={"Timestamp":1740166817466109444,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52440","SessionID":"182650802ccfc7390000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.468179 36900 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52440,hostssl,user=reader] 3325 ={"Timestamp":1740166817468174123,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52440","SessionID":"182650802ccfc7390000000000000001","Duration":5045904}
I250221 19:40:17.497595 36057 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52444,hostssl,user=reader] 3326 ={"Timestamp":1740166817497591548,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52444","SessionID":"182650802eae7b740000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.500066 36978 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52444,hostssl,user=reader] 3327 ={"Timestamp":1740166817500060862,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52444","SessionID":"182650802eae7b740000000000000001","Duration":5560181}
I250221 19:40:17.520363 36950 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52446,hostssl,user=reader] 3328 ={"Timestamp":1740166817520358606,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52446","SessionID":"18265080300d66310000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.522360 36163 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52446,hostssl,user=reader] 3329 ={"Timestamp":1740166817522354095,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52446","SessionID":"18265080300d66310000000000000001","Duration":4855671}
I250221 19:40:17.536291 36060 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52448,hostssl,user=reader] 3330 ={"Timestamp":1740166817536287032,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52448","SessionID":"1826508030fe27b00000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.538142 36058 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52448,hostssl,user=reader] 3331 ={"Timestamp":1740166817538136731,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52448","SessionID":"1826508030fe27b00000000000000001","Duration":4859746}
I250221 19:40:17.552418 36954 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52450,hostssl,user=reader] 3332 ={"Timestamp":1740166817552412983,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52450","SessionID":"1826508031f3fd6a0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.554567 36952 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52450,hostssl,user=reader] 3333 ={"Timestamp":1740166817554561326,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52450","SessionID":"1826508031f3fd6a0000000000000001","Duration":5173656}
I250221 19:40:17.568307 36956 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52452,hostssl,user=reader] 3334 ={"Timestamp":1740166817568303369,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52452","SessionID":"1826508032e6e9380000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.570907 36894 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52452,hostssl,user=reader] 3335 ={"Timestamp":1740166817570901868,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52452","SessionID":"1826508032e6e9380000000000000001","Duration":5594047}
I250221 19:40:17.592879 37043 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52454,hostssl,user=reader] 3336 ={"Timestamp":1740166817592874392,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52454","SessionID":"182650803459bc460000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.594992 36897 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52454,hostssl,user=reader] 3337 ={"Timestamp":1740166817594987257,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52454","SessionID":"182650803459bc460000000000000001","Duration":5377006}
I250221 19:40:17.613550 31574 sql/catalog/lease/lease.go:480  [-] 3338  waiting for 1 leases to expire: desc=[{users 4 24}]
I250221 19:40:17.619073 36715 sql/catalog/lease/descriptor_state.go:135  [n1] 3339  new lease: tabledesc.immutable: {ID: 4, Version: 25, ModificationTime: "1740166817.575477222,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250221 19:40:17.619273 37003 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52456,hostssl,user=reader] 3340 ={"Timestamp":1740166817619269309,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52456","SessionID":"1826508035a61a9a0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250221 19:40:17.622746 36720 sql/catalog/lease/descriptor_state.go:135  [n1] 3341  new lease: tabledesc.immutable: {ID: 20, Version: 25, ModificationTime: "1740166817.575477222,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250221 19:40:17.624422 37075 sql/catalog/lease/descriptor_state.go:135  [n1] 3342  new lease: tabledesc.immutable: {ID: 23, Version: 25, ModificationTime: "1740166817.575477222,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 19:40:17.625731 37092 sql/catalog/lease/descriptor_state.go:135  [n1] 3343  new lease: tabledesc.immutable: {ID: 33, Version: 25, ModificationTime: "1740166817.575477222,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250221 19:40:17.628106 37063 sql/catalog/lease/descriptor_state.go:135  [n1] 3344  new lease: tabledesc.immutable: {ID: 44, Version: 25, ModificationTime: "1740166817.575477222,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250221 19:40:17.630158 37070 sql/catalog/lease/descriptor_state.go:135  [n1] 3345  new lease: tabledesc.immutable: {ID: 104, Version: 24, ModificationTime: "1740166817.575477222,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f589f0]

goroutine 37123 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400818e7a8?, {0x68907b8, 0x400f3c59a0})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a1ed20?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x400a5c4d00?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x68907b8, 0x400f3c59a0})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x68907b8?, 0x400f3c59a0?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400d15d290, {0x68907b8, 0x400f3c59a0}, {0x40110168b8, 0x11}, 0x400818ef40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x4006aafbc0, {0x68937f8?, 0x400dd19180?}, 0x400c371a20, {0x4001f09018, 0x3}, {0x400d15d290?, 0x54?}, 0x401060fb30)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14695
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18900135?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18900135?buildTab=artifacts#/) on master @ [40fef854cdff7ac52f825c0365ba335570022e13](https://github.com/cockroachdb/cockroach/commits/40fef854cdff7ac52f825c0365ba335570022e13):


```
I250222 00:26:53.182944 27829 sql/catalog/lease/descriptor_state.go:135  [n1] 2577  new lease: tabledesc.immutable: {ID: 4, Version: 14, ModificationTime: "1740184013.149968670,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 00:26:53.183785 18785 sql/catalog/lease/lease.go:480  [-] 2578  waiting for 1 leases to expire: desc=[{users 4 13}]
I250222 00:26:53.184258 27834 sql/catalog/lease/descriptor_state.go:135  [n1] 2579  new lease: tabledesc.immutable: {ID: 20, Version: 14, ModificationTime: "1740184013.149968670,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 00:26:53.185881 27839 sql/catalog/lease/descriptor_state.go:135  [n1] 2580  new lease: tabledesc.immutable: {ID: 23, Version: 14, ModificationTime: "1740184013.149968670,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 00:26:53.187166 28099 sql/catalog/lease/descriptor_state.go:135  [n1] 2581  new lease: tabledesc.immutable: {ID: 33, Version: 14, ModificationTime: "1740184013.149968670,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 00:26:53.188890 28119 sql/catalog/lease/descriptor_state.go:135  [n1] 2582  new lease: tabledesc.immutable: {ID: 44, Version: 14, ModificationTime: "1740184013.149968670,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 00:26:53.190391 28105 sql/catalog/lease/descriptor_state.go:135  [n1] 2583  new lease: tabledesc.immutable: {ID: 104, Version: 13, ModificationTime: "1740184013.149968670,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250222 00:26:53.191757 28110 sql/catalog/lease/descriptor_state.go:135  [n1] 2584  new lease: tabledesc.immutable: {ID: 107, Version: 13, ModificationTime: "1740184013.149968670,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 00:26:53.192955 28075 sql/catalog/lease/descriptor_state.go:135  [n1] 2585  new lease: tabledesc.immutable: {ID: 109, Version: 13, ModificationTime: "1740184013.149968670,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 00:26:53.203050 28341 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52504,hostssl,user=reader] 2586 ={"Timestamp":1740184013203045473,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52504","SessionID":"18266023de4c677c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 00:26:53.208089 28077 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52504,hostssl,user=reader] 2587 ={"Timestamp":1740184013208084338,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52504","SessionID":"18266023de4c677c0000000000000001","Duration":13135473}
I250222 00:26:53.729642 27615 sql/catalog/lease/descriptor_state.go:135  [n1] 2588  new lease: tabledesc.immutable: {ID: 4, Version: 15, ModificationTime: "1740184013.700597451,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 00:26:53.730645 18785 sql/catalog/lease/lease.go:480  [-] 2589  waiting for 1 leases to expire: desc=[{users 4 14}]
I250222 00:26:53.731139 28854 sql/catalog/lease/descriptor_state.go:135  [n1] 2590  new lease: tabledesc.immutable: {ID: 20, Version: 15, ModificationTime: "1740184013.700597451,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 00:26:53.732504 27215 sql/catalog/lease/descriptor_state.go:135  [n1] 2591  new lease: tabledesc.immutable: {ID: 23, Version: 15, ModificationTime: "1740184013.700597451,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 00:26:53.734415 28885 sql/catalog/lease/descriptor_state.go:135  [n1] 2592  new lease: tabledesc.immutable: {ID: 33, Version: 15, ModificationTime: "1740184013.700597451,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 00:26:53.736599 28890 sql/catalog/lease/descriptor_state.go:135  [n1] 2593  new lease: tabledesc.immutable: {ID: 44, Version: 15, ModificationTime: "1740184013.700597451,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 00:26:53.737744 28858 sql/catalog/lease/descriptor_state.go:135  [n1] 2594  new lease: tabledesc.immutable: {ID: 104, Version: 14, ModificationTime: "1740184013.700597451,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250222 00:26:53.739104 28862 sql/catalog/lease/descriptor_state.go:135  [n1] 2595  new lease: tabledesc.immutable: {ID: 107, Version: 14, ModificationTime: "1740184013.700597451,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 00:26:53.740297 28843 sql/catalog/lease/descriptor_state.go:135  [n1] 2596  new lease: tabledesc.immutable: {ID: 109, Version: 14, ModificationTime: "1740184013.700597451,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 00:26:53.750797 29239 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52506,hostssl,user=reader] 2597 ={"Timestamp":1740184013750792843,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52506","SessionID":"18266023fef0a3190000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 00:26:53.754411 28845 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:52506,hostssl,user=reader] 2598 ={"Timestamp":1740184013754405579,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:52506","SessionID":"18266023fef0a3190000000000000001","Duration":11822386}
I250222 00:26:54.343381 18785 sql/catalog/lease/lease.go:480  [-] 2599  waiting for 1 leases to expire: desc=[{users 4 15}]
I250222 00:26:54.343593 29595 sql/catalog/lease/descriptor_state.go:135  [n1] 2600  new lease: tabledesc.immutable: {ID: 4, Version: 16, ModificationTime: "1740184014.304214971,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 00:26:54.345159 29615 sql/catalog/lease/descriptor_state.go:135  [n1] 2601  new lease: tabledesc.immutable: {ID: 20, Version: 16, ModificationTime: "1740184014.304214971,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 00:26:54.346421 29601 sql/catalog/lease/descriptor_state.go:135  [n1] 2602  new lease: tabledesc.immutable: {ID: 23, Version: 16, ModificationTime: "1740184014.304214971,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 00:26:54.347942 29638 sql/catalog/lease/descriptor_state.go:135  [n1] 2603  new lease: tabledesc.immutable: {ID: 33, Version: 16, ModificationTime: "1740184014.304214971,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 00:26:54.349511 29339 sql/catalog/lease/descriptor_state.go:135  [n1] 2604  new lease: tabledesc.immutable: {ID: 44, Version: 16, ModificationTime: "1740184014.304214971,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 00:26:54.351142 29644 sql/catalog/lease/descriptor_state.go:135  [n1] 2605  new lease: tabledesc.immutable: {ID: 104, Version: 15, ModificationTime: "1740184014.304214971,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f589f0]

goroutine 29666 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400e2047a8?, {0x68907b8, 0x4008db0230})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a1ed20?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x9ccdd00?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x68907b8, 0x4008db0230})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x68907b8?, 0x4008db0230?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x4010157440, {0x68907b8, 0x4008db0230}, {0x4007e4a030, 0x11}, 0x400e204f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400d44cb40, {0x68937f8?, 0x400c83ea00?}, 0x40057de420, {0x400a90e308, 0x3}, {0x4010157440?, 0x54?}, 0x401122c270)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 15014
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18900335?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18900335?buildTab=artifacts#/) on master @ [c471a724f99a5add3ccefe07995251760d0e2212](https://github.com/cockroachdb/cockroach/commits/c471a724f99a5add3ccefe07995251760d0e2212):


```
I250222 04:27:12.885700 19686 sql/catalog/lease/lease.go:480  [-] 2855  waiting for 1 leases to expire: desc=[{t1 107 11}]
I250222 04:27:12.885696 27592 sql/catalog/lease/descriptor_state.go:135  [n1] 2856  new lease: tabledesc.immutable: {ID: 109, Version: 12, ModificationTime: "1740198432.734515674,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 04:27:14.301463 19686 sql/catalog/lease/lease.go:480  [-] 2857  waiting for 1 leases to expire: desc=[{users 4 13}]
I250222 04:27:14.302121 28803 sql/catalog/lease/descriptor_state.go:135  [n1] 2858  new lease: tabledesc.immutable: {ID: 4, Version: 14, ModificationTime: "1740198434.214096858,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 04:27:14.317155 19686 sql/catalog/lease/lease.go:480  [-] 2859  waiting for 1 leases to expire: desc=[{table_statistics 20 13}]
I250222 04:27:14.321056 28777 sql/catalog/lease/descriptor_state.go:135  [n1] 2860  new lease: tabledesc.immutable: {ID: 20, Version: 14, ModificationTime: "1740198434.214096858,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 04:27:14.322609 27854 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:36954,hostssl,user=reader] 2861 ={"Timestamp":1740198434322604933,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:36954","SessionID":"18266d418c1960b40000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 04:27:14.325019 27852 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:36954,hostssl,user=reader] 2862 ={"Timestamp":1740198434325014071,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:36954","SessionID":"18266d418c1960b40000000000000001","Duration":8960965}
I250222 04:27:14.327180 19686 sql/catalog/lease/lease.go:480  [-] 2863  waiting for 1 leases to expire: desc=[{role_members 23 13}]
I250222 04:27:14.328883 28782 sql/catalog/lease/descriptor_state.go:135  [n1] 2864  new lease: tabledesc.immutable: {ID: 23, Version: 14, ModificationTime: "1740198434.214096858,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 04:27:14.336399 19686 sql/catalog/lease/lease.go:480  [-] 2865  waiting for 1 leases to expire: desc=[{role_options 33 13}]
I250222 04:27:14.342274 28852 sql/catalog/lease/descriptor_state.go:135  [n1] 2866  new lease: tabledesc.immutable: {ID: 33, Version: 14, ModificationTime: "1740198434.214096858,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 04:27:14.347331 28857 sql/catalog/lease/descriptor_state.go:135  [n1] 2867  new lease: tabledesc.immutable: {ID: 44, Version: 14, ModificationTime: "1740198434.214096858,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 04:27:14.351869 19686 sql/catalog/lease/lease.go:480  [-] 2868  waiting for 1 leases to expire: desc=[{database_role_settings 44 13}]
I250222 04:27:14.356538 28862 sql/catalog/lease/descriptor_state.go:135  [n1] 2869  new lease: tabledesc.immutable: {ID: 104, Version: 13, ModificationTime: "1740198434.214096858,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250222 04:27:14.362444 19686 sql/catalog/lease/lease.go:480  [-] 2870  waiting for 1 leases to expire: desc=[{sq1 104 12}]
I250222 04:27:14.364551 28898 sql/catalog/lease/descriptor_state.go:135  [n1] 2871  new lease: tabledesc.immutable: {ID: 107, Version: 13, ModificationTime: "1740198434.214096858,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 04:27:14.366120 28810 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:36956,hostssl,user=reader] 2872 ={"Timestamp":1740198434366115585,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:36956","SessionID":"18266d418e8f79000000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 04:27:14.369538 19686 sql/catalog/lease/lease.go:480  [-] 2873  waiting for 1 leases to expire: desc=[{t1 107 12}]
I250222 04:27:14.380691 28903 sql/catalog/lease/descriptor_state.go:135  [n1] 2874  new lease: tabledesc.immutable: {ID: 109, Version: 13, ModificationTime: "1740198434.214096858,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 04:27:14.394107 28819 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:36956,hostssl,user=reader] 2875 ={"Timestamp":1740198434394102057,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:36956","SessionID":"18266d418e8f79000000000000000001","Duration":36754536}
I250222 04:27:15.462659 29582 sql/catalog/lease/descriptor_state.go:135  [n1] 2876  new lease: tabledesc.immutable: {ID: 4, Version: 15, ModificationTime: "1740198435.404531387,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 04:27:15.464816 29569 sql/catalog/lease/descriptor_state.go:135  [n1] 2877  new lease: tabledesc.immutable: {ID: 20, Version: 15, ModificationTime: "1740198435.404531387,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 04:27:15.474003 29622 sql/catalog/lease/descriptor_state.go:135  [n1] 2878  new lease: tabledesc.immutable: {ID: 23, Version: 15, ModificationTime: "1740198435.404531387,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 04:27:15.477163 29627 sql/catalog/lease/descriptor_state.go:135  [n1] 2879  new lease: tabledesc.immutable: {ID: 33, Version: 15, ModificationTime: "1740198435.404531387,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 04:27:15.488196 29632 sql/catalog/lease/descriptor_state.go:135  [n1] 2880  new lease: tabledesc.immutable: {ID: 44, Version: 15, ModificationTime: "1740198435.404531387,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 04:27:15.492301 29652 sql/catalog/lease/descriptor_state.go:135  [n1] 2881  new lease: tabledesc.immutable: {ID: 104, Version: 14, ModificationTime: "1740198435.404531387,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250222 04:27:15.497534 29375 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:36958,hostssl,user=reader] 2882 ={"Timestamp":1740198435497529909,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:36958","SessionID":"18266d41d19dacb30000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 04:27:15.494114 29657 sql/catalog/lease/descriptor_state.go:135  [n1] 2883  new lease: tabledesc.immutable: {ID: 107, Version: 14, ModificationTime: "1740198435.404531387,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f58a90]

goroutine 29662 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400eece7a8?, {0x6890958, 0x400182bc20})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a1ee60?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x400bcfd560?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x6890958, 0x400182bc20})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x6890958?, 0x400182bc20?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x40071cf950, {0x6890958, 0x400182bc20}, {0x4004a45548, 0x11}, 0x400eecef40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x4001762060, {0x6893998?, 0x400504cf00?}, 0x40058ad1e0, {0x400b1bb1d0, 0x3}, {0x40071cf950?, 0x54?}, 0x400bbc89a8)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 15494
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18903632?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18903632?buildTab=artifacts#/) on master @ [3de70cd8a25ed906838a8d5cde65c8e244149698](https://github.com/cockroachdb/cockroach/commits/3de70cd8a25ed906838a8d5cde65c8e244149698):


```
I250222 16:04:29.179987 42890 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41416,hostssl,user=reader] 3497 ={"Timestamp":1740240269179983172,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41416","SessionID":"1826934dfc15ca780000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.181834 42888 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41416,hostssl,user=reader] 3498 ={"Timestamp":1740240269181829072,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41416","SessionID":"1826934dfc15ca780000000000000001","Duration":5499619}
I250222 16:04:29.194587 42893 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41418,hostssl,user=reader] 3499 ={"Timestamp":1740240269194583218,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41418","SessionID":"1826934dfcffa2ce0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.196221 42808 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41418,hostssl,user=reader] 3500 ={"Timestamp":1740240269196215940,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41418","SessionID":"1826934dfcffa2ce0000000000000001","Duration":4561381}
I250222 16:04:29.209327 42941 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41420,hostssl,user=reader] 3501 ={"Timestamp":1740240269209323717,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41420","SessionID":"1826934dfddff9830000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.211100 42369 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41420,hostssl,user=reader] 3502 ={"Timestamp":1740240269211095714,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41420","SessionID":"1826934dfddff9830000000000000001","Duration":4739183}
I250222 16:04:29.229997 41126 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41422,hostssl,user=reader] 3503 ={"Timestamp":1740240269229992668,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41422","SessionID":"1826934dff1ce0c60000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.231712 41173 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41422,hostssl,user=reader] 3504 ={"Timestamp":1740240269231708216,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41422","SessionID":"1826934dff1ce0c60000000000000001","Duration":4582799}
I250222 16:04:29.246027 42958 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41424,hostssl,user=reader] 3505 ={"Timestamp":1740240269246023240,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41424","SessionID":"1826934e000094560000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.247895 42810 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41424,hostssl,user=reader] 3506 ={"Timestamp":1740240269247890104,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41424","SessionID":"1826934e000094560000000000000001","Duration":5841987}
I250222 16:04:29.260664 41176 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41426,hostssl,user=reader] 3507 ={"Timestamp":1740240269260660368,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41426","SessionID":"1826934e00eebd100000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.262528 42959 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41426,hostssl,user=reader] 3508 ={"Timestamp":1740240269262523213,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41426","SessionID":"1826934e00eebd100000000000000001","Duration":4867418}
I250222 16:04:29.277465 41178 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41428,hostssl,user=reader] 3509 ={"Timestamp":1740240269277460704,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41428","SessionID":"1826934e01f0516c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.279392 43006 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41428,hostssl,user=reader] 3510 ={"Timestamp":1740240269279387958,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41428","SessionID":"1826934e01f0516c0000000000000001","Duration":4851290}
I250222 16:04:29.299512 41180 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41430,hostssl,user=reader] 3511 ={"Timestamp":1740240269299507592,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41430","SessionID":"1826934e03406a470000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.301457 42961 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41430,hostssl,user=reader] 3512 ={"Timestamp":1740240269301452168,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41430","SessionID":"1826934e03406a470000000000000001","Duration":4888462}
I250222 16:04:29.319210 43058 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41432,hostssl,user=reader] 3513 ={"Timestamp":1740240269319206225,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41432","SessionID":"1826934e046bc6f10000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.321308 43038 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41432,hostssl,user=reader] 3514 ={"Timestamp":1740240269321303245,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41432","SessionID":"1826934e046bc6f10000000000000001","Duration":5120240}
I250222 16:04:29.336646 43078 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41434,hostssl,user=reader] 3515 ={"Timestamp":1740240269336641756,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41434","SessionID":"1826934e057254df0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.338922 43076 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41434,hostssl,user=reader] 3516 ={"Timestamp":1740240269338917059,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41434","SessionID":"1826934e057254df0000000000000001","Duration":5527877}
I250222 16:04:29.359684 43081 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41436,hostssl,user=reader] 3517 ={"Timestamp":1740240269359677754,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41436","SessionID":"1826934e066db8e30000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:04:29.362589 43079 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41436,hostssl,user=reader] 3518 ={"Timestamp":1740240269362583688,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41436","SessionID":"1826934e066db8e30000000000000001","Duration":12719646}
I250222 16:04:29.365787 43060 sql/catalog/lease/descriptor_state.go:135  [n1] 3519  new lease: tabledesc.immutable: {ID: 4, Version: 31, ModificationTime: "1740240269.328925149,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 16:04:29.366870 31908 sql/catalog/lease/lease.go:480  [-] 3520  waiting for 1 leases to expire: desc=[{users 4 30}]
I250222 16:04:29.367085 43106 sql/catalog/lease/descriptor_state.go:135  [n1] 3521  new lease: tabledesc.immutable: {ID: 20, Version: 31, ModificationTime: "1740240269.328925149,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 16:04:29.368327 43111 sql/catalog/lease/descriptor_state.go:135  [n1] 3522  new lease: tabledesc.immutable: {ID: 23, Version: 31, ModificationTime: "1740240269.328925149,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 16:04:29.370210 43117 sql/catalog/lease/descriptor_state.go:135  [n1] 3523  new lease: tabledesc.immutable: {ID: 33, Version: 31, ModificationTime: "1740240269.328925149,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 16:04:29.371698 43122 sql/catalog/lease/descriptor_state.go:135  [n1] 3524  new lease: tabledesc.immutable: {ID: 44, Version: 31, ModificationTime: "1740240269.328925149,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 16:04:29.373470 43127 sql/catalog/lease/descriptor_state.go:135  [n1] 3525  new lease: tabledesc.immutable: {ID: 104, Version: 30, ModificationTime: "1740240269.328925149,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f58a90]

goroutine 43132 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400fed37a8?, {0x6890958, 0x4010ff60a0})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a1ee60?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x400cb26f80?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x6890958, 0x4010ff60a0})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x6890958?, 0x4010ff60a0?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400e1fcea0, {0x6890958, 0x4010ff60a0}, {0x400c323218, 0x11}, 0x400fed3f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400dc15b60, {0x6893998?, 0x400e873e00?}, 0x4010c5d760, {0x400c780830, 0x3}, {0x400e1fcea0?, 0x54?}, 0x4010c59ce0)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 15012
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18904098?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18904098?buildTab=artifacts#/) on master @ [3de70cd8a25ed906838a8d5cde65c8e244149698](https://github.com/cockroachdb/cockroach/commits/3de70cd8a25ed906838a8d5cde65c8e244149698):


```
I250222 16:22:14.765579 24280 sql/catalog/lease/descriptor_state.go:135  [n1] 2497  new lease: tabledesc.immutable: {ID: 104, Version: 10, ModificationTime: "1740241334.738003949,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250222 16:22:14.766412 24290 sql/catalog/lease/descriptor_state.go:135  [n1] 2498  new lease: tabledesc.immutable: {ID: 107, Version: 10, ModificationTime: "1740241334.738003949,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 16:22:14.767310 24212 sql/catalog/lease/descriptor_state.go:135  [n1] 2499  new lease: tabledesc.immutable: {ID: 109, Version: 10, ModificationTime: "1740241334.738003949,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 16:22:14.775667 24616 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38974,hostssl,user=reader] 2500 ={"Timestamp":1740241334775662626,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38974","SessionID":"1826944616616b400000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:22:14.778587 24214 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38974,hostssl,user=reader] 2501 ={"Timestamp":1740241334778583640,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38974","SessionID":"1826944616616b400000000000000001","Duration":9201732}
I250222 16:22:15.053454 22368 13@kv/kvserver/replica_command.go:481  [T1,Vsystem,n1,split,s1,r97/1:/Tenant/1{0/Table/…-1}] 2502  initiating a split of this range at key /Tenant/10/Table/104 [r98] (span config)
I250222 16:22:15.053471 24936 13@kv/kvserver/replica_command.go:481  [T1,Vsystem,n1,split,s1,r96/1:/Tenant/1{1/Table/…-2}] 2503  initiating a split of this range at key /Tenant/11/Table/104 [r99] (span config)
I250222 16:22:15.054437 22368 13@kv/kvserver/batcheval/cmd_end_transaction.go:1289  [T1,Vsystem,n1,s1,r97/1:/Tenant/1{0/Table/…-1}] 2504  falling back to accurate stats computation because there are no pre-split LHS stats (or they're empty)
I250222 16:22:15.054513 24936 13@kv/kvserver/batcheval/cmd_end_transaction.go:1289  [T1,Vsystem,n1,s1,r96/1:/Tenant/1{1/Table/…-2}] 2505  falling back to accurate stats computation because there are no pre-split LHS stats (or they're empty)
I250222 16:22:15.055218 11498 2@kv/kvserver/replica_proposal.go:620  [T1,Vsystem,n1,s1,r96/1:/Tenant/1{1/Table/…-2},raft] 2506  new range lease repl=(n1,s1):1 seq=1 start=0,0 epo=1 min-exp=0,0 pro=1740241326.083765350,0 acq=Request following repl=(n1,s1):1 seq=1 start=0,0 epo=1 min-exp=0,0 pro=1740241326.083765350,0 acq=Request
I250222 16:22:15.138525 25059 sql/catalog/lease/descriptor_state.go:135  [n1] 2507  new lease: tabledesc.immutable: {ID: 4, Version: 12, ModificationTime: "1740241335.115106445,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 16:22:15.139206 17886 sql/catalog/lease/lease.go:480  [-] 2508  waiting for 1 leases to expire: desc=[{users 4 11}]
I250222 16:22:15.139491 25082 sql/catalog/lease/descriptor_state.go:135  [n1] 2509  new lease: tabledesc.immutable: {ID: 20, Version: 12, ModificationTime: "1740241335.115106445,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 16:22:15.140516 24982 sql/catalog/lease/descriptor_state.go:135  [n1] 2510  new lease: tabledesc.immutable: {ID: 23, Version: 12, ModificationTime: "1740241335.115106445,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 16:22:15.141533 25041 sql/catalog/lease/descriptor_state.go:135  [n1] 2511  new lease: tabledesc.immutable: {ID: 33, Version: 12, ModificationTime: "1740241335.115106445,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 16:22:15.142494 25115 sql/catalog/lease/descriptor_state.go:135  [n1] 2512  new lease: tabledesc.immutable: {ID: 44, Version: 12, ModificationTime: "1740241335.115106445,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 16:22:15.143515 25008 sql/catalog/lease/descriptor_state.go:135  [n1] 2513  new lease: tabledesc.immutable: {ID: 104, Version: 11, ModificationTime: "1740241335.115106445,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250222 16:22:15.144348 25143 sql/catalog/lease/descriptor_state.go:135  [n1] 2514  new lease: tabledesc.immutable: {ID: 107, Version: 11, ModificationTime: "1740241335.115106445,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 16:22:15.145204 25146 sql/catalog/lease/descriptor_state.go:135  [n1] 2515  new lease: tabledesc.immutable: {ID: 109, Version: 11, ModificationTime: "1740241335.115106445,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250222 16:22:15.155307 25438 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38976,hostssl,user=reader] 2516 ={"Timestamp":1740241335155303312,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38976","SessionID":"182694462d0000dd0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250222 16:22:15.156837 25352 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:38976,hostssl,user=reader] 2517 ={"Timestamp":1740241335156833455,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:38976","SessionID":"182694462d0000dd0000000000000001","Duration":7959751}
I250222 16:22:15.519180 25661 sql/catalog/lease/descriptor_state.go:135  [n1] 2518  new lease: tabledesc.immutable: {ID: 4, Version: 13, ModificationTime: "1740241335.497008068,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250222 16:22:15.519968 17886 sql/catalog/lease/lease.go:480  [-] 2519  waiting for 1 leases to expire: desc=[{users 4 12}]
I250222 16:22:15.520274 25858 sql/catalog/lease/descriptor_state.go:135  [n1] 2520  new lease: tabledesc.immutable: {ID: 20, Version: 13, ModificationTime: "1740241335.497008068,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250222 16:22:15.521286 25876 sql/catalog/lease/descriptor_state.go:135  [n1] 2521  new lease: tabledesc.immutable: {ID: 23, Version: 13, ModificationTime: "1740241335.497008068,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 16:22:15.522514 25882 sql/catalog/lease/descriptor_state.go:135  [n1] 2522  new lease: tabledesc.immutable: {ID: 33, Version: 13, ModificationTime: "1740241335.497008068,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250222 16:22:15.523448 25840 sql/catalog/lease/descriptor_state.go:135  [n1] 2523  new lease: tabledesc.immutable: {ID: 44, Version: 13, ModificationTime: "1740241335.497008068,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250222 16:22:15.524393 25757 sql/catalog/lease/descriptor_state.go:135  [n1] 2524  new lease: tabledesc.immutable: {ID: 104, Version: 12, ModificationTime: "1740241335.497008068,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250222 16:22:15.525350 25938 sql/catalog/lease/descriptor_state.go:135  [n1] 2525  new lease: tabledesc.immutable: {ID: 107, Version: 12, ModificationTime: "1740241335.497008068,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f58a90]

goroutine 25909 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400e58d7a8?, {0x6890958, 0x4009dd6780})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a1ee60?, 0x9bd2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x400d79b080?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x6890958, 0x4009dd6780})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x6890958?, 0x4009dd6780?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x4010641050, {0x6890958, 0x4009dd6780}, {0x400f763dd0, 0x11}, 0x400e58df40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400192f3b0, {0x6893998?, 0x4008033e00?}, 0x4008b758c0, {0x4009ddb0a8, 0x3}, {0x4010641050?, 0x54?}, 0x40091f43f0)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14445
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18911315?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18911315?buildTab=artifacts#/) on master @ [cbb292c487ea7413772a5c3c349f93467264c190](https://github.com/cockroachdb/cockroach/commits/cbb292c487ea7413772a5c3c349f93467264c190):


```
I250224 00:38:09.536064 32452 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54124,hostssl,user=reader] 2741 ={"Timestamp":1740357489536059773,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54124","SessionID":"1826fdea79b1cd740000000000000001","Duration":4911258}
I250224 00:38:09.549259 32487 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54126,hostssl,user=reader] 2742 ={"Timestamp":1740357489549254299,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54126","SessionID":"1826fdea7a97f7750000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.551221 32279 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54126,hostssl,user=reader] 2743 ={"Timestamp":1740357489551216251,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54126","SessionID":"1826fdea7a97f7750000000000000001","Duration":4983825}
I250224 00:38:09.567441 32490 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54128,hostssl,user=reader] 2744 ={"Timestamp":1740357489567436621,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54128","SessionID":"1826fdea7bad4d690000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.569520 32488 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54128,hostssl,user=reader] 2745 ={"Timestamp":1740357489569514570,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54128","SessionID":"1826fdea7bad4d690000000000000001","Duration":5106809}
I250224 00:38:09.584455 32512 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54130,hostssl,user=reader] 2746 ={"Timestamp":1740357489584451235,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54130","SessionID":"1826fdea7cb1b0460000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.586462 32510 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54130,hostssl,user=reader] 2747 ={"Timestamp":1740357489586456842,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54130","SessionID":"1826fdea7cb1b0460000000000000001","Duration":4984592}
I250224 00:38:09.607402 32288 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54132,hostssl,user=reader] 2748 ={"Timestamp":1740357489607398439,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54132","SessionID":"1826fdea7e0f76640000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.609900 32286 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54132,hostssl,user=reader] 2749 ={"Timestamp":1740357489609894693,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54132","SessionID":"1826fdea7e0f76640000000000000001","Duration":5499304}
I250224 00:38:09.623167 32549 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54134,hostssl,user=reader] 2750 ={"Timestamp":1740357489623162502,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54134","SessionID":"1826fdea7f00c79e0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.625330 32547 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54134,hostssl,user=reader] 2751 ={"Timestamp":1740357489625324623,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54134","SessionID":"1826fdea7f00c79e0000000000000001","Duration":5114430}
I250224 00:38:09.638028 32552 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54136,hostssl,user=reader] 2752 ={"Timestamp":1740357489638023416,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54136","SessionID":"1826fdea7fe12b260000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.640369 32480 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54136,hostssl,user=reader] 2753 ={"Timestamp":1740357489640364097,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54136","SessionID":"1826fdea7fe12b260000000000000001","Duration":5448219}
I250224 00:38:09.663474 32518 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54138,hostssl,user=reader] 2754 ={"Timestamp":1740357489663469815,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54138","SessionID":"1826fdea8165a52c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.667543 32481 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54138,hostssl,user=reader] 2755 ={"Timestamp":1740357489667537388,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54138","SessionID":"1826fdea8165a52c0000000000000001","Duration":7162550}
I250224 00:38:09.680887 32611 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54140,hostssl,user=reader] 2756 ={"Timestamp":1740357489680883416,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54140","SessionID":"1826fdea826bc6a90000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.682781 32520 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54140,hostssl,user=reader] 2757 ={"Timestamp":1740357489682776086,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54140","SessionID":"1826fdea826bc6a90000000000000001","Duration":5221863}
I250224 00:38:09.694844 32608 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54142,hostssl,user=reader] 2758 ={"Timestamp":1740357489694839909,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54142","SessionID":"1826fdea8345cb390000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.697085 32606 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54142,hostssl,user=reader] 2759 ={"Timestamp":1740357489697078717,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54142","SessionID":"1826fdea8345cb390000000000000001","Duration":5235689}
I250224 00:38:09.710963 32643 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54144,hostssl,user=reader] 2760 ={"Timestamp":1740357489710958578,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54144","SessionID":"1826fdea842bb2710000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.713150 32529 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54144,hostssl,user=reader] 2761 ={"Timestamp":1740357489713144179,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54144","SessionID":"1826fdea842bb2710000000000000001","Duration":6235165}
I250224 00:38:09.728201 32645 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54146,hostssl,user=reader] 2762 ={"Timestamp":1740357489728196053,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54146","SessionID":"1826fdea854265100000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.731449 32635 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54146,hostssl,user=reader] 2763 ={"Timestamp":1740357489731443921,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54146","SessionID":"1826fdea854265100000000000000001","Duration":6268382}
I250224 00:38:09.750917 32640 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54148,hostssl,user=reader] 2764 ={"Timestamp":1740357489750913389,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54148","SessionID":"1826fdea869c79290000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 00:38:09.756083 32638 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:54148,hostssl,user=reader] 2765 ={"Timestamp":1740357489756075499,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:54148","SessionID":"1826fdea869c79290000000000000001","Duration":8220921}
I250224 00:38:09.765858 32556 sql/catalog/lease/descriptor_state.go:135  [n1] 2766  new lease: tabledesc.immutable: {ID: 4, Version: 21, ModificationTime: "1740357489.733397753,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250224 00:38:09.766807 30834 sql/catalog/lease/lease.go:480  [-] 2767  waiting for 1 leases to expire: desc=[{users 4 20}]
I250224 00:38:09.767559 32561 sql/catalog/lease/descriptor_state.go:135  [n1] 2768  new lease: tabledesc.immutable: {ID: 20, Version: 21, ModificationTime: "1740357489.733397753,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250224 00:38:09.770713 32678 sql/catalog/lease/descriptor_state.go:135  [n1] 2769  new lease: tabledesc.immutable: {ID: 23, Version: 21, ModificationTime: "1740357489.733397753,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f58a90]

goroutine 32684 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400ab537a8?, {0x689ccf8, 0x400a989ea0})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a27c40?, 0x9be2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x400a67d6a0?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x689ccf8, 0x400a989ea0})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x689ccf8?, 0x400a989ea0?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x40040ac7e0, {0x689ccf8, 0x400a989ea0}, {0x4005a29af0, 0x10}, 0x400ab53f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x4007fb2660, {0x689fd38?, 0x4009f663c0?}, 0x400bc278c0, {0x54bdcbc, 0x2}, {0x40040ac7e0?, 0x54?}, 0x4005aa60c0)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14310
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18915553?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18915553?buildTab=artifacts#/) on master @ [881952f2ef5a75c2e5f5b0808aba20213bb34d63](https://github.com/cockroachdb/cockroach/commits/881952f2ef5a75c2e5f5b0808aba20213bb34d63):


```
I250224 15:03:16.507537 41912 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41368,hostssl,user=reader] 3760 ={"Timestamp":1740409396507532741,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41368","SessionID":"18272d20026725c50000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.509721 41910 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41368,hostssl,user=reader] 3761 ={"Timestamp":1740409396509716287,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41368","SessionID":"18272d20026725c50000000000000001","Duration":5210642}
I250224 15:03:16.524014 42007 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41370,hostssl,user=reader] 3762 ={"Timestamp":1740409396524010003,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41370","SessionID":"18272d200362d7830000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.542702 42025 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41370,hostssl,user=reader] 3763 ={"Timestamp":1740409396542696015,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41370","SessionID":"18272d200362d7830000000000000001","Duration":21693991}
I250224 15:03:16.559362 41915 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41372,hostssl,user=reader] 3764 ={"Timestamp":1740409396559357906,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41372","SessionID":"18272d20057aec4c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.561607 41913 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41372,hostssl,user=reader] 3765 ={"Timestamp":1740409396561601724,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41372","SessionID":"18272d20057aec4c0000000000000001","Duration":5467865}
I250224 15:03:16.585426 42114 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41374,hostssl,user=reader] 3766 ={"Timestamp":1740409396585421293,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41374","SessionID":"18272d20070625760000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.588330 42086 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41374,hostssl,user=reader] 3767 ={"Timestamp":1740409396588323808,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41374","SessionID":"18272d20070625760000000000000001","Duration":6288347}
I250224 15:03:16.620837 40640 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41376,hostssl,user=reader] 3768 ={"Timestamp":1740409396620831849,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41376","SessionID":"18272d200927b75a0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.623283 42073 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41376,hostssl,user=reader] 3769 ={"Timestamp":1740409396623276830,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41376","SessionID":"18272d200927b75a0000000000000001","Duration":5486982}
I250224 15:03:16.641832 42106 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41378,hostssl,user=reader] 3770 ={"Timestamp":1740409396641827217,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41378","SessionID":"18272d200a65ae480000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.644387 42074 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41378,hostssl,user=reader] 3771 ={"Timestamp":1740409396644382008,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41378","SessionID":"18272d200a65ae480000000000000001","Duration":5753752}
I250224 15:03:16.683872 42131 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41380,hostssl,user=reader] 3772 ={"Timestamp":1740409396683868227,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41380","SessionID":"18272d200be395150000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.686797 42076 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41380,hostssl,user=reader] 3773 ={"Timestamp":1740409396686790903,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41380","SessionID":"18272d200be395150000000000000001","Duration":23134222}
I250224 15:03:16.700829 42160 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41382,hostssl,user=reader] 3774 ={"Timestamp":1740409396700824847,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41382","SessionID":"18272d200dec2c2b0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.703185 42077 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41382,hostssl,user=reader] 3775 ={"Timestamp":1740409396703179422,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41382","SessionID":"18272d200dec2c2b0000000000000001","Duration":5405721}
I250224 15:03:16.718220 42181 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41384,hostssl,user=reader] 3776 ={"Timestamp":1740409396718214682,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41384","SessionID":"18272d200ef5256d0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.720493 42170 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41384,hostssl,user=reader] 3777 ={"Timestamp":1740409396720487453,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41384","SessionID":"18272d200ef5256d0000000000000001","Duration":5348092}
I250224 15:03:16.736284 42111 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41386,hostssl,user=reader] 3778 ={"Timestamp":1740409396736279034,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41386","SessionID":"18272d201009d4550000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.741544 42109 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41386,hostssl,user=reader] 3779 ={"Timestamp":1740409396741535584,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41386","SessionID":"18272d201009d4550000000000000001","Duration":8264021}
I250224 15:03:16.758515 31164 sql/catalog/lease/lease.go:480  [-] 3780  waiting for 1 leases to expire: desc=[{users 4 30}]
I250224 15:03:16.760729 42144 sql/catalog/lease/descriptor_state.go:135  [n1] 3781  new lease: tabledesc.immutable: {ID: 4, Version: 31, ModificationTime: "1740409396.714158988,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250224 15:03:16.762724 42213 sql/catalog/lease/descriptor_state.go:135  [n1] 3782  new lease: tabledesc.immutable: {ID: 20, Version: 31, ModificationTime: "1740409396.714158988,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250224 15:03:16.763253 42194 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41388,hostssl,user=reader] 3783 ={"Timestamp":1740409396763248556,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41388","SessionID":"18272d2011678f600000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:03:16.764358 42227 sql/catalog/lease/descriptor_state.go:135  [n1] 3784  new lease: tabledesc.immutable: {ID: 23, Version: 31, ModificationTime: "1740409396.714158988,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 15:03:16.766210 42186 sql/catalog/lease/descriptor_state.go:135  [n1] 3785  new lease: tabledesc.immutable: {ID: 33, Version: 31, ModificationTime: "1740409396.714158988,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 15:03:16.766614 42143 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:41388,hostssl,user=reader] 3786 ={"Timestamp":1740409396766609331,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:41388","SessionID":"18272d2011678f600000000000000001","Duration":10417613}
I250224 15:03:16.767578 42091 sql/catalog/lease/descriptor_state.go:135  [n1] 3787  new lease: tabledesc.immutable: {ID: 44, Version: 31, ModificationTime: "1740409396.714158988,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250224 15:03:16.768827 42096 sql/catalog/lease/descriptor_state.go:135  [n1] 3788  new lease: tabledesc.immutable: {ID: 104, Version: 30, ModificationTime: "1740409396.714158988,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f58a90]

goroutine 42190 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x40117167a8?, {0x689ccf8, 0x4010632910})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a27c40?, 0x9be2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x4012d5f5c0?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x689ccf8, 0x4010632910})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x689ccf8?, 0x4010632910?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400fbf0630, {0x689ccf8, 0x4010632910}, {0x4011d52888, 0x11}, 0x4011716f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x4011572240, {0x689fd38?, 0x40135aebe0?}, 0x4012340000, {0x400ebe1d50, 0x3}, {0x400fbf0630?, 0x54?}, 0x4010fb66c0)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14801
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18916062?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18916062?buildTab=artifacts#/) on master @ [b9c6bb6a0570d1022b34c0be699cb1c9cf10126e](https://github.com/cockroachdb/cockroach/commits/b9c6bb6a0570d1022b34c0be699cb1c9cf10126e):


```
I250224 15:31:03.444620 22675 13@kv/kvserver/batcheval/cmd_end_transaction.go:1289  [T1,Vsystem,n1,s1,r87/1:/Tenant/1{0/Table/…-1}] 2739  falling back to accurate stats computation because there are no pre-split LHS stats (or they're empty)
I250224 15:31:03.446975 11613 2@kv/kvserver/replica_proposal.go:620  [T1,Vsystem,n1,s1,r98/1:/Tenant/1{0/Table/…-1},raft] 2740  new range lease repl=(n1,s1):1 seq=1 start=0,0 term=6 min-exp=1740411069.446585482,0 pro=1740411063.446585482,0 acq=Request following repl=(n1,s1):1 seq=1 start=0,0 exp=1740411069.444927765,0 pro=1740411057.575237281,0 acq=Request
I250224 15:31:03.447205 22998 13@kv/kvserver/replica_command.go:481  [T1,Vsystem,n1,split,s1,r97/1:/Tenant/1{1/Table/…-2}] 2741  initiating a split of this range at key /Tenant/11/Table/104 [r99] (span config)
I250224 15:31:03.448916 22998 13@kv/kvserver/batcheval/cmd_end_transaction.go:1289  [T1,Vsystem,n1,s1,r97/1:/Tenant/1{1/Table/…-2}] 2742  falling back to accurate stats computation because there are no pre-split LHS stats (or they're empty)
I250224 15:31:03.451026 11629 kv/kvserver/kvserverbase/forced_error.go:87  [T1,Vsystem,n1,s1,r99/1:/Tenant/1{1/Table/…-2},raft] 2743  rejected lease request repl=(n1,s1):1 seq=1 start=0,0 exp=1740411069.450416563,0 pro=1740411063.450416563,0 acq=Request; current lease repl=(n1,s1):1 seq=1 start=0,0 exp=1740411069.450416563,0 pro=1740411063.450416563,0 acq=Request; err: cannot replace lease repl=(n1,s1):1 seq=1 start=0,0 exp=1740411069.450416563,0 pro=1740411063.450416563,0 acq=Request with repl=(n1,s1):1 seq=1 start=0,0 exp=1740411069.450416563,0 pro=1740411063.450416563,0 acq=Request: proposed under invalid lease
I250224 15:31:03.467829 22411 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51312,hostssl,user=reader] 2744 ={"Timestamp":1740411063467824330,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51312","SessionID":"18272ea41eb013fa0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:31:03.473821 22272 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51312,hostssl,user=reader] 2745 ={"Timestamp":1740411063473816855,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51312","SessionID":"18272ea41eb013fa0000000000000001","Duration":47457933}
I250224 15:31:03.923638 11632 2@kv/kvserver/replica_proposal.go:626  [T1,Vsystem,n1,s1,r99/1:/Tenant/1{1/Table/…-2},raft] 2746  new range lease repl=(n1,s1):1 seq=1 start=0,0 term=6 min-exp=1740411069.923342154,0 pro=1740411063.923342154,0 acq=Request promoted from repl=(n1,s1):1 seq=1 start=0,0 exp=1740411069.450416563,0 pro=1740411063.450416563,0 acq=Request
I250224 15:31:04.146290 22278 sql/catalog/lease/descriptor_state.go:135  [n1] 2747  new lease: tabledesc.immutable: {ID: 4, Version: 9, ModificationTime: "1740411064.118604817,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250224 15:31:04.147634 18403 sql/catalog/lease/lease.go:480  [-] 2748  waiting for 1 leases to expire: desc=[{users 4 8}]
I250224 15:31:04.148173 23414 sql/catalog/lease/descriptor_state.go:135  [n1] 2749  new lease: tabledesc.immutable: {ID: 20, Version: 9, ModificationTime: "1740411064.118604817,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250224 15:31:04.150005 23419 sql/catalog/lease/descriptor_state.go:135  [n1] 2750  new lease: tabledesc.immutable: {ID: 23, Version: 9, ModificationTime: "1740411064.118604817,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 15:31:04.152024 23425 sql/catalog/lease/descriptor_state.go:135  [n1] 2751  new lease: tabledesc.immutable: {ID: 33, Version: 9, ModificationTime: "1740411064.118604817,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 15:31:04.154292 23430 sql/catalog/lease/descriptor_state.go:135  [n1] 2752  new lease: tabledesc.immutable: {ID: 44, Version: 9, ModificationTime: "1740411064.118604817,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250224 15:31:04.168400 18403 sql/catalog/lease/lease.go:480  [-] 2753  waiting for 1 leases to expire: desc=[{sq1 104 7}]
I250224 15:31:04.172164 23435 sql/catalog/lease/descriptor_state.go:135  [n1] 2754  new lease: tabledesc.immutable: {ID: 104, Version: 8, ModificationTime: "1740411064.118604817,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250224 15:31:04.174185 23408 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51314,hostssl,user=reader] 2755 ={"Timestamp":1740411064174181993,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51314","SessionID":"18272ea44a6c12230000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 15:31:04.181729 21425 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51314,hostssl,user=reader] 2756 ={"Timestamp":1740411064181723626,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51314","SessionID":"18272ea44a6c12230000000000000001","Duration":21624152}
I250224 15:31:04.182188 23440 sql/catalog/lease/descriptor_state.go:135  [n1] 2757  new lease: tabledesc.immutable: {ID: 107, Version: 8, ModificationTime: "1740411064.118604817,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250224 15:31:04.183851 23449 sql/catalog/lease/descriptor_state.go:135  [n1] 2758  new lease: tabledesc.immutable: {ID: 109, Version: 8, ModificationTime: "1740411064.118604817,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true}, {ID: 2, TypeID: 20, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
I250224 15:31:04.184519 18403 sql/catalog/lease/lease.go:480  [-] 2759  waiting for 1 leases to expire: desc=[{t2 109 7}]
I250224 15:31:04.738128 24244 sql/catalog/lease/descriptor_state.go:135  [n1] 2760  new lease: tabledesc.immutable: {ID: 4, Version: 10, ModificationTime: "1740411064.709832582,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250224 15:31:04.739094 18403 sql/catalog/lease/lease.go:480  [-] 2761  waiting for 1 leases to expire: desc=[{users 4 9}]
I250224 15:31:04.739499 23734 sql/catalog/lease/descriptor_state.go:135  [n1] 2762  new lease: tabledesc.immutable: {ID: 20, Version: 10, ModificationTime: "1740411064.709832582,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250224 15:31:04.740812 23739 sql/catalog/lease/descriptor_state.go:135  [n1] 2763  new lease: tabledesc.immutable: {ID: 23, Version: 10, ModificationTime: "1740411064.709832582,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 15:31:04.742272 24032 sql/catalog/lease/descriptor_state.go:135  [n1] 2764  new lease: tabledesc.immutable: {ID: 33, Version: 10, ModificationTime: "1740411064.709832582,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 15:31:04.743563 24176 sql/catalog/lease/descriptor_state.go:135  [n1] 2765  new lease: tabledesc.immutable: {ID: 44, Version: 10, ModificationTime: "1740411064.709832582,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250224 15:31:04.744631 24277 sql/catalog/lease/descriptor_state.go:135  [n1] 2766  new lease: tabledesc.immutable: {ID: 104, Version: 9, ModificationTime: "1740411064.709832582,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250224 15:31:04.746434 24249 sql/catalog/lease/descriptor_state.go:135  [n1] 2767  new lease: tabledesc.immutable: {ID: 107, Version: 9, ModificationTime: "1740411064.709832582,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f58a90]

goroutine 24254 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400997f7a8?, {0x689c1f8, 0x4003a87310})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a273c0?, 0x9be2750?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x4010e1c9c0?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x689c1f8, 0x4003a87310})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x689c1f8?, 0x4003a87310?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400a9c5a70, {0x689c1f8, 0x4003a87310}, {0x4002ee7a58, 0x11}, 0x400997ff40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x400a7c9440, {0x689f238?, 0x400fdd7040?}, 0x4003fac580, {0x4004320488, 0x3}, {0x400a9c5a70?, 0x54?}, 0x4002cdf200)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14704
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/catalog/replication/replication_test.TestReaderCatalogTSAdvance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18916896?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18916896?buildTab=artifacts#/) on master @ [4215a94873a014c9aee49b043bc419cb93bd83d5](https://github.com/cockroachdb/cockroach/commits/4215a94873a014c9aee49b043bc419cb93bd83d5):


```
I250224 16:47:46.653753 32933 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51654,hostssl,user=reader] 2992 ={"Timestamp":1740415666653748098,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51654","SessionID":"182732d3e41be2b80000000000000001","Duration":5238648}
I250224 16:47:46.666278 32936 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51656,hostssl,user=reader] 2993 ={"Timestamp":1740415666666273977,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51656","SessionID":"182732d3e4fb64f80000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.668244 32880 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51656,hostssl,user=reader] 2994 ={"Timestamp":1740415666668239168,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51656","SessionID":"182732d3e4fb64f80000000000000001","Duration":5081251}
I250224 16:47:46.680568 32972 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51658,hostssl,user=reader] 2995 ={"Timestamp":1740415666680563695,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51658","SessionID":"182732d3e5d96da30000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.682804 32970 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51658,hostssl,user=reader] 2996 ={"Timestamp":1740415666682798947,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51658","SessionID":"182732d3e5d96da30000000000000001","Duration":5090385}
I250224 16:47:46.707156 32976 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51662,hostssl,user=reader] 2997 ={"Timestamp":1740415666707151843,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51662","SessionID":"182732d3e76efa110000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.709217 32974 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51662,hostssl,user=reader] 2998 ={"Timestamp":1740415666709209397,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51662","SessionID":"182732d3e76efa110000000000000001","Duration":4922807}
I250224 16:47:46.726135 33011 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51664,hostssl,user=reader] 2999 ={"Timestamp":1740415666726130069,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51664","SessionID":"182732d3e848edfb0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.728823 32979 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51664,hostssl,user=reader] 3000 ={"Timestamp":1740415666728818248,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51664","SessionID":"182732d3e848edfb0000000000000001","Duration":10247643}
I250224 16:47:46.743748 33014 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51666,hostssl,user=reader] 3001 ={"Timestamp":1740415666743743814,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51666","SessionID":"182732d3e99cb78c0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.745989 33012 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51666,hostssl,user=reader] 3002 ={"Timestamp":1740415666745983712,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51666","SessionID":"182732d3e99cb78c0000000000000001","Duration":5145072}
I250224 16:47:46.759479 32174 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51668,hostssl,user=reader] 3003 ={"Timestamp":1740415666759474932,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51668","SessionID":"182732d3ea8dc9400000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.761457 33015 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51668,hostssl,user=reader] 3004 ={"Timestamp":1740415666761451184,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51668","SessionID":"182732d3ea8dc9400000000000000001","Duration":4813559}
I250224 16:47:46.777090 33017 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51670,hostssl,user=reader] 3005 ={"Timestamp":1740415666777085600,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51670","SessionID":"182732d3eb9463c00000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.779463 32911 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51670,hostssl,user=reader] 3006 ={"Timestamp":1740415666779457604,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51670","SessionID":"182732d3eb9463c00000000000000001","Duration":5609877}
I250224 16:47:46.811436 32981 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51672,hostssl,user=reader] 3007 ={"Timestamp":1740415666811430901,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51672","SessionID":"182732d3ed4166cb0000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.813457 32176 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51672,hostssl,user=reader] 3008 ={"Timestamp":1740415666813452358,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51672","SessionID":"182732d3ed4166cb0000000000000001","Duration":11489052}
I250224 16:47:46.826416 33091 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51674,hostssl,user=reader] 3009 ={"Timestamp":1740415666826411425,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51674","SessionID":"182732d3ee7a03a00000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.828594 32177 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51674,hostssl,user=reader] 3010 ={"Timestamp":1740415666828588073,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51674","SessionID":"182732d3ee7a03a00000000000000001","Duration":6137465}
I250224 16:47:46.842777 33087 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51676,hostssl,user=reader] 3011 ={"Timestamp":1740415666842771112,"EventType":"client_authentication_ok","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51676","SessionID":"182732d3ef7af6260000000000000001","Transport":"hostssl","User":"reader","SystemIdentity":"reader","Method":"cert-password"}
I250224 16:47:46.864451 33107 4@util/log/event_log.go:39  [T11,Vdest,n1,client=127.0.0.1:51676,hostssl,user=reader] 3012 ={"Timestamp":1740415666864444373,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:51676","SessionID":"182732d3ef7af6260000000000000001","Duration":25154362}
I250224 16:47:46.866712 28791 sql/catalog/lease/descriptor_state.go:135  [n1] 3013  new lease: tabledesc.immutable: {ID: 4, Version: 21, ModificationTime: "1740415666.827405665,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 17, Null: true}, {ID: 3, TypeID: 16, Null: false, HasDefault: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 4, Families: [{ID: 0, Columns: [1, 4]}, {ID: 2, Columns: [2]}, {ID: 3, Columns: [3]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}], StoreColumns: [2, 3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1]}]}
I250224 16:47:46.867882 31248 sql/catalog/lease/lease.go:480  [-] 3014  waiting for 1 leases to expire: desc=[{users 4 20}]
I250224 16:47:46.868041 32230 sql/catalog/lease/descriptor_state.go:135  [n1] 3015  new lease: tabledesc.immutable: {ID: 20, Version: 21, ModificationTime: "1740415666.827405665,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 13, Columns: [{ID: 1, TypeID: 20, Null: false}, {ID: 2, TypeID: 20, Null: false, HasDefault: true}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 1016, Null: false}, {ID: 5, TypeID: 1114, Null: false, HasDefault: true}, {ID: 6, TypeID: 20, Null: false}, {ID: 7, TypeID: 20, Null: false}, {ID: 8, TypeID: 20, Null: false}, {ID: 9, TypeID: 17, Null: true}, {ID: 10, TypeID: 20, Null: false, HasDefault: true}, {ID: 11, TypeID: 25, Null: true}, {ID: 12, TypeID: 20, Null: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}]}
I250224 16:47:46.869774 32992 sql/catalog/lease/descriptor_state.go:135  [n1] 3016  new lease: tabledesc.immutable: {ID: 23, Version: 21, ModificationTime: "1740415666.827405665,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 6, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 16, Null: false}, {ID: 4, TypeID: 26, Null: false}, {ID: 5, TypeID: 26, Null: false}], NextFamilyID: 6, Families: [{ID: 0, Columns: [1, 2]}, {ID: 3, Columns: [3]}, {ID: 4, Columns: [4]}, {ID: 5, Columns: [5]}], PrimaryIndex: 1, NextIndexID: 7, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4, 5]}, {ID: 2, Unique: false, KeyColumns: [{ID: 1, Dir: ASC}], KeySuffixColumns: [2]}, {ID: 3, Unique: false, KeyColumns: [{ID: 2, Dir: ASC}], KeySuffixColumns: [1]}, {ID: 4, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 5, Unique: false, KeyColumns: [{ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}, {ID: 6, Unique: true, KeyColumns: [{ID: 4, Dir: ASC}, {ID: 5, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 16:47:46.871373 33095 sql/catalog/lease/descriptor_state.go:135  [n1] 3017  new lease: tabledesc.immutable: {ID: 33, Version: 21, ModificationTime: "1740415666.827405665,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 25, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 25, Null: true}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: false, KeyColumns: [{ID: 4, Dir: ASC}], KeySuffixColumns: [1, 2]}]}
I250224 16:47:46.872808 33139 sql/catalog/lease/descriptor_state.go:135  [n1] 3018  new lease: tabledesc.immutable: {ID: 44, Version: 21, ModificationTime: "1740415666.827405665,0", ParentID: 1, ParentSchemaID: 29, State: PUBLIC, NextColumnID: 5, Columns: [{ID: 1, TypeID: 26, Null: false}, {ID: 2, TypeID: 25, Null: false}, {ID: 3, TypeID: 1009, Null: false}, {ID: 4, TypeID: 26, Null: false}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3, 4]}], PrimaryIndex: 1, NextIndexID: 3, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 2, Dir: ASC}], StoreColumns: [3, 4]}, {ID: 2, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}, {ID: 4, Dir: ASC}], KeySuffixColumns: [2], StoreColumns: [3]}]}
I250224 16:47:46.873898 33110 sql/catalog/lease/descriptor_state.go:135  [n1] 3019  new lease: tabledesc.immutable: {ID: 104, Version: 20, ModificationTime: "1740415666.827405665,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, Sequence: true, NextColumnID: 2, Columns: [{ID: 1, TypeID: 20, Null: false}], NextFamilyID: 0, PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 1, Dir: ASC}]}]}
I250224 16:47:46.875125 33068 sql/catalog/lease/descriptor_state.go:135  [n1] 3020  new lease: tabledesc.immutable: {ID: 107, Version: 20, ModificationTime: "1740415666.827405665,0", ParentID: 100, ParentSchemaID: 101, State: PUBLIC, NextColumnID: 4, Columns: [{ID: 1, TypeID: 20, Null: true, HasDefault: true, UsesSequenceIDs: [104]}, {ID: 2, TypeID: 100105, Null: true}, {ID: 3, TypeID: 20, Null: false, Hidden: true, HasDefault: true}], NextFamilyID: 1, Families: [{ID: 0, Columns: [1, 2, 3]}], PrimaryIndex: 1, NextIndexID: 2, Indexes: [{ID: 1, Unique: true, KeyColumns: [{ID: 3, Dir: ASC}], StoreColumns: [1, 2]}]}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1f59700]

goroutine 33034 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400ea317a8?, {0x689e4b8, 0x400f59fc20})
	pkg/util/stop/stopper.go:227 +0x68
panic({0x4a28a00?, 0x9be2760?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).getSessionID(0x2?)
	pkg/sql/catalog/lease/descriptor_version_state.go:115 +0xc0
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.acquireNodeLease.func1({0x689e4b8, 0x400f59fc20})
	pkg/sql/catalog/lease/lease.go:859 +0xf0
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall.func1({0x689e4b8?, 0x400f59fc20?})
	pkg/util/syncutil/singleflight/singleflight.go:384 +0x38
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask(0x400a17f710, {0x689e4b8, 0x400f59fc20}, {0x400b014600, 0x11}, 0x400ea31f40)
	pkg/util/stop/stopper.go:317 +0x100
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0x4009af6d50, {0x68a1538?, 0x400a1635e0?}, 0x4013191b80, {0x400d5e8e58, 0x3}, {0x400a17f710?, 0x54?}, 0x400ac2d350)
	pkg/util/syncutil/singleflight/singleflight.go:383 +0x21c
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan in goroutine 14753
	pkg/util/syncutil/singleflight/singleflight.go:353 +0x4b8
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReaderCatalogTSAdvance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/catalog/lease/descriptor_state.go
+++ b/pkg/sql/catalog/lease/descriptor_state.go
@@ -181,13 +181,12 @@ func newDescriptorVersionState(
 			version: int(desc.GetVersion()),
 		}
 		descState.mu.lease.sessionID = session.ID().UnsafeBytes()
-		descState.mu.session = session
-
 		if buildutil.CrdbTestBuild && !expiration.IsEmpty() {
 			panic(errors.AssertionFailedf("expiration should always be empty for "+
 				"session based leases (got: %s on Desc: %s(%d))", expiration.String(), desc.GetName(), desc.GetID()))
 		}
 	}
+	descState.mu.session = session
 	descState.mu.expiration = expiration
 
 	return descState

--- a/pkg/sql/catalog/lease/descriptor_version_state.go
+++ b/pkg/sql/catalog/lease/descriptor_version_state.go
@@ -60,9 +60,7 @@ type descriptorVersionState struct {
 		// when the version isn't associated with a lease.
 		expiration hlc.Timestamp
 
-		// The session that was used to acquire this descriptor version, which is
-		// only populated when the session based leasing mode is *at least* dual
-		// write.
+		// The session that was used to acquire this descriptor version.
 		session sqlliveness.Session
 
 		refcount int

--- a/pkg/sql/catalog/lease/lease.go
+++ b/pkg/sql/catalog/lease/lease.go
@@ -774,8 +774,14 @@ func (m *Manager) readOlderVersionForTimestamp(
 
 // Insert descriptor versions. The versions provided are not in
 // any particular order.
-func (m *Manager) insertDescriptorVersions(id descpb.ID, versions []historicalDescriptor) {
+func (m *Manager) insertDescriptorVersions(
+	ctx context.Context, id descpb.ID, versions []historicalDescriptor,
+) error {
 	t := m.findDescriptorState(id, false /* create */)
+	session, err := m.storage.livenessProvider.Session(ctx)
+	if err != nil {
+		return err
+	}
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	for i := range versions {
@@ -785,9 +791,10 @@ func (m *Manager) insertDescriptorVersions(id descpb.ID, versions []historicalDe
 		existingVersion := t.mu.active.findVersion(versions[i].desc.GetVersion())
 		if existingVersion == nil {
 			t.mu.active.insert(
-				newDescriptorVersionState(t, versions[i].desc, versions[i].expiration, nil, nil, false))
+				newDescriptorVersionState(t, versions[i].desc, versions[i].expiration, session, nil, false))
 		}
 	}
+	return nil
 }
 
 // AcquireFreshestFromStore acquires a new lease from the store and
@@ -1471,8 +1478,10 @@ func (m *Manager) Acquire(
 			if errRead != nil {
 				return nil, errRead
 			}
-			m.insertDescriptorVersions(id, versions)
-
+			errRead = m.insertDescriptorVersions(ctx, id, versions)
+			if errRead != nil {
+				return nil, errRead
+			}
 		default:
 			return nil, err
 		}

